horace_core/sqw/@sqw/get_nearest_pixels.m

```diff
@@ -1,34 +1,65 @@
-function [xp, ipix] = get_nearest_pixels (win, xp_in)
-% Get the the indicies of the nearest pixels to an array of points along the plot axes
+function [ok, ipix] = get_nearest_pixels (win, xp)
+% Find the nearest pixel to each point of an array along the plot axes
 %
-%   >> xp, ipix] = get_nearest_pixels (win, xp_in)
+%   >> [ok, ipix] = get_nearest_pixels (win, xp)
+%
+% The algorithm determines which bin the point is in, and then finds the
+% nearest pixel in that bin. If a point is in a bin without pixels, or lies
+% outside the sqw object, then it is ignored
 %
 % Input:
 % ------
-%   win     sqw object
-%   xp_in   Row vector length np, or n x np array, where np is the number of 
-%          plot axes, and n the number of points
+%   win     A single sqw object
+%
+%   xp      Point or points for which to determine the nearest pixel;
+%          array size  [n, np], where np is the number of plot axes and
+%          n the number of points
 %
 % Output:
 % -------
-%   xp      Row vector length np, or n x np array, where n is now the 
-%          number of retained points, that is, ones which lie in the range
-%          of the data and which lie in bins that have at least one pixel
-%   ipix    Column vector, length n, with the corresponding indicies of the
-%          nearest pixel to each point.
+%   ok      Logical column vector length equal to the number of points
+%          where
+%           - 1 indicates the point has a nearest pixel in the bin in
+%            which the point sits
+%           - 0 indicates the point lies in a bin without any contributing
+%            pixels, or the point lies outside the range of the sqw dataset
+%
+%   ipix    Column vector with the indicies of the nearest pixel to each
+%          for which ok is true; that is for those points given by xp(ok,:)
+%
 %           The nearest pixel is defined as that which has minimum distance
 %          in terms of step size along each axis from the point defined by
 %          xp along the plot axes and median coordinate of the pixels along
 %          the integration axes.
 
 
+if numel(win)>1
+    error('HORACE:get_nearest_pixels:invalid_argument',...
+        'Only a single instance of an sqw object can be passed.')
+end
+
 nd = numel(win.data.pax);
 sz = size(win.data.npix);
-np = size(xp_in,1);
-if size(xp_in,2)~=nd
-    error('Number of coordinates in the point does not match dimensions of sqw object')
+np = size(xp,1);
+if numel(size(xp)) ~= 2
+    error('HORACE:get_nearest_pixels:invalid_argument',...
+        'Coordinate array must be a row vector or two-dimensional array.')
+elseif size(xp, 2) ~= nd
+    error('HORACE:get_nearest_pixels:invalid_argument',...
+        'Number of coordinates in the point does not match the dimensions of the sqw object.')
+end
+
+% Catch case of no query points on input
+% ----------------------------------------
+if size(xp, 1)==0
+    ok = false(0,1);
```

Comment:
You have `size(xp, 1)` as the variable `np`

Reply:
well spotted. I hadn't.